Userland/DynamicLoader/main.cpp

```diff
@@ -121,8 +126,15 @@ void _entry(int argc, char** argv, char** envp)
     }
     VERIFY(at_random_found);
 
+    // Make sure compiler won't move any functions calls above __stack_chk_guard initialization even
+    // if their definitions somehow become available.
+    optimizer_fence();
```

Comment:
:thinking: Is this the same as `atomic_thread_fence` or `atomic_signal_fence` from the STL? Maybe we could move it to AK later. and/or use it for related pthread features.